services/webhook/tools/retry_backoff_helper.py

```
@@ -0,0 +1,24 @@
+import time
+
+
+def retry_with_backoff(fn, attempts: int = 3, base_delay: float = 0.5):
+    """Call `fn` with exponential backoff between attempts."""
+    last_error: Exception | None = None
+    for attempt in range(attempts):
+        try:
+            return fn()
+        except OSError as e:
+            last_error = e
+            time.sleep(base_delay * (2 ** attempt))
```

Ruff (0.15.21)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)